spec/01-unit/21-dns-client/02-client_spec.lua

```diff
@@ -1767,7 +1773,9 @@ describe("[DNS client]", function()
           touch = 0,
           expire = gettime() + 10,
         }
-        sleep(0.5) -- wait before we return the results
+        -- wait before we return the results
+        -- `+ 2` s ensures that the semaphore:wait() expires
+        sleep(timeout/1000 + 2)
```

Comment:
Also would it be possible to use some kind of waiter function pattern that would check against a predicate every _interval_ ?

Reply:
> Also would it be possible to use some kind of waiter function pattern that would check against a predicate every interval ?

If the waiting of the test case happens under two endpoint of C/S model, like that a client wait for server to complete some tasks, your advice is the best way to do. We need to be especially care of handling the synchronization problem between two different process to prevent the flakiness.

But here we only run the DNS query and wait the result in one process. So if this test case fails, there must be some pressing problem happening in the DNS library.

Reply:
TL;DR: it is a unit test, the chance of flakiness is much lower because client runs much faster than the entire Gateway